examples/verify_tx.rs

```diff
@@ -83,11 +83,11 @@ fn main() {
         0xa9, 0x14, 0x92, 0x09, 0xa8, 0xf9, 0x0c, 0x58, 0x4b, 0xb5, 0x97, 0x4d, 0x58, 0x68, 0x72,
         0x49, 0xe5, 0x32, 0xde, 0x59, 0xf4, 0xbc, 0x87,
     ]);
-
+    let wit = transaction.input[0].witness.to_vec();
     let mut interpreter = miniscript::Interpreter::from_txdata(
```

Comment:
What do you think to change the parameter type in `from_txdata` (and in `inner::from_txdata`) to `&Witness` avoiding the step of converting to vec? 

If I am looking correctly it's used only for iteration and we have `Witness::iter` which doesn't allocate

Reply:
Yes, this is an already planned change in a follow-up PR. Wanted to keep things simple here.

Reply:
Does it make sense for psbt FINAL_SCRIPT_WITNESS to be `Witness` type instead of `Vec<Vec<u8>>`

Reply:
Added a follow up commit and discovered that rust-miniscript needs https://github.com/rust-bitcoin/rust-bitcoin/pull/774 for a cleaner non-allocating API.